ignition-server/controllers/local_ignitionprovider.go

```diff
@@ -0,0 +1,434 @@
+		}
+		return data, nil
+	}()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pull secret: %w", err)
+	}
+
+	// Fetch the bootstrap kubeconfig contents
+	bootstrapKubeConfig, err := func() ([]byte, error) {
+		secret := &corev1.Secret{}
+		if err := p.Client.Get(ctx, client.ObjectKey{Namespace: p.Namespace, Name: "bootstrap-kubeconfig"}, secret); err != nil {
+			return nil, fmt.Errorf("failed to get bootstrap kubeconfig secret: %w", err)
+		}
+		data, exists := secret.Data["kubeconfig"]
+		if !exists {
+			return nil, fmt.Errorf("bootstrap kubeconfig secret missing kubeconfig key")
+		}
+		return data, nil
+	}()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get bootstrap kubeconfig: %w", err)
+	}
+
+	// Fetch the MCS config
+	mcsConfig := &corev1.ConfigMap{}
+	if err := p.Client.Get(ctx, client.ObjectKey{Namespace: p.Namespace, Name: "machine-config-server"}, mcsConfig); err != nil {
+		return nil, fmt.Errorf("failed to get machine-config-server configmap: %w", err)
+	}
+
+	// Look up the release image metadata
+	images, err := func() (map[string]string, error) {
+		img, err := p.ReleaseProvider.Lookup(ctx, releaseImage, pullSecret)
+		if err != nil {
+			return nil, fmt.Errorf("failed to look up release image metadata: %w", err)
+		}
+		return img.ComponentImages(), nil
+	}()
+
+	mcoImage, hasMcoImage := images["machine-config-operator"]
+	if !hasMcoImage {
+		return nil, fmt.Errorf("release image does not contain machine-config-operator")
+	}
+	log.Info("discovered mco image", "image", mcoImage)
+
+	// Set up the base working directory
+	workDir, err := ioutil.TempDir(p.WorkDir, "get-payload")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create working directory: %w", err)
+	}
+	defer func() {
+		if err := os.RemoveAll(workDir); err != nil {
+			log.Error(err, "failed to delete working directory", "dir", workDir)
+		}
+	}()
+	log.Info("created working directory", "dir", workDir)
+
+	// Prepare all the working subdirectories
+	binDir := filepath.Join(workDir, "bin")
+	mcoBaseDir := filepath.Join(workDir, "mco")
+	mccBaseDir := filepath.Join(workDir, "mcc")
+	mcsBaseDir := filepath.Join(workDir, "mcs")
+	configDir := filepath.Join(workDir, "config")
+	for _, dir := range []string{binDir, mcoBaseDir, mccBaseDir, mcsBaseDir, configDir} {
+		if err := os.Mkdir(dir, 0755); err != nil {
+			return nil, fmt.Errorf("failed to make directory %s: %w", dir, err)
+		}
+	}
+
+	// Write out the custom config to the MCC directory
+	if err := os.WriteFile(filepath.Join(mccBaseDir, "custom.yaml"), []byte(customConfig), 0644); err != nil {
+		return nil, fmt.Errorf("failed to write mcc config: %w", err)
+	}
+	// Write out the bootstrap kubeconfig to the MCS directory
+	if err := os.WriteFile(filepath.Join(mcsBaseDir, "kubeconfig"), bootstrapKubeConfig, 0644); err != nil {
+		return nil, fmt.Errorf("failed to write bootstrap kubeconfig: %w", err)
+	}
+	// Extract MCS config files into the config directory
```

Comment:
this needs to occur after the cmd := exec.CommandContext(ctx, filepath.Join(binDir, "machine-config-operator" command execution and content copy or else the proper machine config pools will not be used

Reply:
At time of this writing I think the proper spot for this is 294:
https://github.com/openshift/hypershift/pull/1214/files#diff-7cc71524ad99a2a526cceabd327a67c2767dc0c49e5c4aa8629ca88b0d205072R294

Reply:
Fixed